### PR TITLE
Add purchase task E2E test

### LINKED ISSUE #32326: WooCommerce package.json `@woocommerce/admin-e2e-tests` dependency still points to release number
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**
WooCommerce package.json `@woocommerce/admin-e2e-tests` dependency still points to release number instead of the local available package `workspace:*`.

### Testing instructions
- Update or add a new test in packages/admin-e2e-tests/specs
- Run the build
- Now run the E2E tests by following the instructions [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e/README.md)
- Notice how the updates aren't being reflected

### PR DESCRIPTION
**Pull request by louwie17**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds new purchase task E2E test in order for this to work I had to fix the `admin-e2e-tests` dependency and point it to our local workspace now, and I also added the WooCommerce reset plugin.

Closes #32326 .

### How to test the changes in this Pull Request:

1. Make sure the E2E tests pass in the Github action
2. Pull this branch and run the E2E locally by following these instructions - https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e/README.md the whole test suite take a while so you could just run the new tests -> `pnpm exec wc-e2e test:e2e-dev ./tests/e2e/specs/admin-tasks/purchase.test.js` (inside the `plugins/woocommerce` folder)

### Changelog entry

> Add E2E test for the purchase task in the task list.

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/admin-e2e-tests/src/fixtures/plugins.ts [hunk 1]
```diff
@@ -47,15 +47,18 @@ async function deactivateAndDeletePlugin( pluginName: string ) {
...
 	let plugins = await getPlugins();
+	let skippedPlugins = [];
 	const promises = [];
 	for ( const plugin of plugins ) {
 		const splitPluginName = plugin.plugin.split( '/' );
 		const slug = splitPluginName[ 1 ] || splitPluginName[ 0 ];
 		if ( ! except.includes( slug ) ) {
 			promises.push( deactivateAndDeletePlugin( plugin.plugin ) );
+		} else {
+			skippedPlugins.push( slug );
```
Comment (louwie17) [inline packages/js/admin-e2e-tests/src/fixtures/plugins.ts:58]: Not all the same plugins passed into `except` are installed, so make sure we only check the plugins length of the actual installed skipped plugins.

File: plugins/woocommerce/package.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
     "@typescript-eslint/eslint-plugin": "3.10.1",
     "@typescript-eslint/experimental-utils": "3.10.1",
     "@typescript-eslint/parser": "3.10.1",
-    "@woocommerce/admin-e2e-tests": "0.1.1",
+    "@woocommerce/admin-e2e-tests": "workspace:*",
```
Comment (octaedro) [inline plugins/woocommerce/package.json:42]: ❤️

File: packages/js/e2e-environment/docker/wp-cli/initialize.sh [hunk 3]
```diff
@@ -18,6 +18,8 @@ wp user create customer customer@woocommercecoree2etestsuite.com \
 # we cannot create API keys for the API, so we using basic auth, this plugin allows that.
 wp plugin install https://github.com/WP-API/Basic-Auth/archive/master.zip --activate
 
+wp plugin install https://github.com/woocommerce/woocommerce-reset/zipball/trunk/ --activate
```
Comment (octaedro) [inline packages/js/e2e-environment/docker/wp-cli/initialize.sh:22]: Nitpick: Could we add a comment with an explanation for this line?

File: plugins/woocommerce/tests/e2e/docker/initialize.sh [hunk 4]
```diff
@@ -18,6 +18,8 @@ wp user create customer customer@woocommercecoree2etestsuite.com \
 # we cannot create API keys for the API, so we using basic auth, this plugin allows that.
 wp plugin install https://github.com/WP-API/Basic-Auth/archive/master.zip --activate
 
+wp plugin install https://github.com/woocommerce/woocommerce-reset/zipball/trunk/ --activate
```
Comment (octaedro) [inline plugins/woocommerce/tests/e2e/docker/initialize.sh:22]: Same here 🙂

**Review by octaedro — DISMISSED** [review]
Good job @louwie17! This is testing well here and code LGTM. I only left a very small suggestion. I'm going to pre-approve this PR 🟢

**Review by octaedro — APPROVED** [review]
LGTM 🚀

**Pull request merged**